usaspending_api/database_scripts/tableview_generator/tableview_sql_generator.py

```diff
@@ -0,0 +1,417 @@
+"""
+A TableView is
+"""
+import argparse
+import glob
+import hashlib
+import json
+import os
+import subprocess
+
+from uuid import uuid4
+
+# Usage: python tableview_sql_generator.py (from usaspending_api/database_scripts/tableview_generator)
+#        ^--- Will clobber files in usaspending_api/database_scripts/tableviews
+
+"""
+POSTGRES INDEX FORMAT
+    CREATE [ UNIQUE ] INDEX [ name ] ON table_name [ USING method ]
+    ( { column_name | ( expression ) } [ COLLATE collation ]
+        [ opclass ] [ ASC | DESC ] [ NULLS { FIRST | LAST } ] [, ...] )
+    [ WITH ( storage_parameter = value [, ... ] ) ]
+    [ WHERE predicate ]
+
+EXAMPLE SQL DESCRIPTION JSON FILE:
```

Comment:
The SQL description JSON file perhaps follows the matview generator pattern, but I think the readability challenges of JSON throws me. 

If this python module is used to conduct management tasks surrounding the lifecycle of the tableview, could we store the concrete implementation of each tableview's template as a python class instead of a JSON file?

e.g. have a base class that defines all of the methods that are expected for each tableview to define:
- `get_create_table_sql(self)`, `get_select_sql(self)`, `get_delta_update_sql(self)`, etc.

And then the class instance for the concrete tableview is provided to this _factory_ to conduct lifecycle changes to the tableview

I think it's just the unreadability of this all being JSON/dictionary that makes me think there are better ways to store schema definitions and lifecycle functions

Reply:
The decision to repurpose the `matview_sql_generator` was largely due to perceived time constraints and convenience.  I wanted to rework this a bit more, but was more interested in getting something up quickly.  Will definitely take a look at this more closely in our next pass.

Reply:
Fair. Lets get the solution working in terms of data flow. Then we can circle back to optimizing operation of the lifecycle of the tableview(s).